GH1855: Adding support for additional locations.

phillipsj opened the pull request:
Relates to #1855.

I did not add any ARM paths.  I didn't see the best way to handle that as there isn't any explicit ARM flags inside of the environment that I could find.

patriksvensson commented:
@phillipsj You're on fire with all your PR:s 😄 👍 
Would it be possible to add some tests for this as well?

phillipsj commented:
Yep i will get some extra ones added.

On Sun, Oct 8, 2017, 4:29 PM Patrik <notifications@github.com> wrote:

> @phillipsj <https://github.com/phillipsj> You're on fire with all your
> PR:s 😄 👍
> Would it be possible to add some tests for this as well?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cake-build/cake/pull/1868#issuecomment-335036077>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABfthTYL8ZHhIfRtQIiY95uLr1fR70GPks5sqTC_gaJpZM4PxzPj>
> .
>

mholo65 commented on src/Cake.Common/Tools/SignTool/SignToolResolver.cs:
```diff
@@ -67,7 +67,9 @@ private FilePath GetFromDisc()
             {
                 // 64-bit specific paths.
-                files.Add(programFilesPath.Combine(@"Windows Kits\10\bin\x64").CombineWithFilePath("signtool.exe"));
+                files.Add(programFilesPath.Combine(@"Windows Kits10\bin\x64").CombineWithFilePath("signtool.exe"));
```
Was this intentional? Is the folder really named `Windows Kits10`? If yes then same thing should be done on 32-bit side, right?

mholo65 commented on src/Cake.Common/Tools/SignTool/SignToolResolver.cs:
```diff
@@ -67,7 +67,9 @@ private FilePath GetFromDisc()
-                files.Add(programFilesPath.Combine(@"Windows Kits\10\bin\x64").CombineWithFilePath("signtool.exe"));
+                files.Add(programFilesPath.Combine(@"Windows Kits10\bin\x64").CombineWithFilePath("signtool.exe"));
+                files.Add(programFilesPath.Combine(@"Windows Kits\10\bin\10.0.15063.0\x64").CombineWithFilePath("signtool.exe"));
+                files.Add(programFilesPath.Combine(@"Windows Kits\10\bin\10.0.14393.0\x64").CombineWithFilePath("signtool.exe"));
```
Instead of hardcoding versions (which would make us add additional path everytime a new version of Win10 SDK is released) could we instead search using pattern `Windows Kits/10/bin/*/x64/signtool.exe`?

phillipsj commented:
Nice suggestions on both.  Thoughts on ARM support?

On Mon, Oct 9, 2017, 12:37 AM Martin Björkström <notifications@github.com>
wrote:

> *@mholo65* commented on this pull request.
> ------------------------------
>
> In src/Cake.Common/Tools/SignTool/SignToolResolver.cs
> <https://github.com/cake-build/cake/pull/1868#discussion_r143383605>:
>
> > @@ -67,7 +67,9 @@ private FilePath GetFromDisc()
>              if (_environment.Platform.Is64Bit)
>              {
>                  // 64-bit specific paths.
> -                files.Add(programFilesPath.Combine(@"Windows Kits\10\bin\x64").CombineWithFilePath("signtool.exe"));
> +                files.Add(programFilesPath.Combine(@"Windows Kits10\bin\x64").CombineWithFilePath("signtool.
>
> exe"));
> +                files.Add(programFilesPath.Combine(@"Windows Kits\10\bin\10.0.15063.0\x64").CombineWithFilePath("signtool.exe"));
> +                files.Add(programFilesPath.Combine(@"Windows Kits\10\bin\10.0.14393.0\x64").CombineWithFilePath("signtool.exe"));
>
> Instead of hardcoding versions (which would make us add additional path
> everytime a new version of Win10 SDK is released) could we instead search
> using pattern Windows Kits/10/bin/*/x64/signtool.exe?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cake-build/cake/pull/1868#pullrequestreview-67887042>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABfthYW4sq2OoTI1FYGrYVrnlCTDdq-iks5sqaMAgaJpZM4PxzPj>
> .
>

phillipsj commented:
@patriksvensson I have added tests.  @mholo65 I have fixed the typo and I tried adding the wildcard, however, the GetFile on the Filesystem object is throwing an error.  I would have to use something like GetFiles, then pull out the paths that way.  That didn't feel like the best way to handle it.

phillipsj commented:
@mholo65 @patriksvensson Does it look good?  Any changes that I need to make?  See my previous comment about using wildcards.

patriksvensson commented:
Sorry for not getting back to you about this. Been super busy lately but will set aside some time asap for this.

phillipsj commented:
@patriksvensson no worries, just making sure I am not holding it up.

Thealexbarney commented:
Hard-coding the SDK versions doesn't seem like the best method. This PR would already be out-of-date with the latest SDK release. The installed Windows 10 SDK versions are are listed as subkeys of `HKLM\SOFTWARE\Microsoft\Windows Kits\Installed Roots`. Those could be used instead of hard-coded versions.

Thealexbarney commented:
I made my own solution with PR #1925. I wasn't sure if I should opened a new PR or somehow attach it to this one, given that the methods of resolving the SDK are different between the two PRs.

devlead commented:
@Thealexbarney agreed, I'll merge this PR and then you can rebase your PR and I'll review that one.

devlead commented:
@phillipsj your changes have been merged, thanks for your contribution 👍
